src/shared/condition.c

```diff
@@ -958,6 +960,113 @@ static int condition_test_file_is_executable(Condition *c, char **env) {
                 (st.st_mode & 0111));
 }
 
+static int condition_test_psi(Condition *c, char **env) {
+        _cleanup_free_ char *first = NULL, *second = NULL, *pressure_path = NULL;
+        loadavg_t *current, limit;
+        ResourcePressure pressure;
+        const char *p, *value;
+        OrderOperator order;
+        PressureType type;
+        int r;
+
+        assert(c);
+        assert(c->parameter);
+        assert(IN_SET(c->type, CONDITION_MEMORY_PRESSURE, CONDITION_CPU_PRESSURE));
+
+        if (!is_pressure_supported())
+                return log_debug_errno(SYNTHETIC_ERRNO(EOPNOTSUPP), "Pressure Stall Information (PSI) is not supported");
+
+        type = c->type == CONDITION_CPU_PRESSURE ? PRESSURE_TYPE_SOME : PRESSURE_TYPE_FULL;
```

Comment:
cpu.pressure full was added in the past year https://lore.kernel.org/lkml/20201225042248.51737-1-zhouchengming@bytedance.com/. Though it looks like the docs aren't updated. Maybe for consistency switch all of these to "full" or explicitly document that cpu.pressure uses "some"?

Reply:
Didn't know that, thanks, will switch to always use 'full' and fallback to 'some' if not available, and document it

Reply:
Ok, done, first check 'full' and fallback to 'some' on ENODATA